flycheck.el

```diff
@@ -7744,7 +7744,12 @@ Otherwise return the previously used cache directory."
 Look up the directory hierarchy from FILE for a directory
 containing a file that matches REGEXP."
   (locate-dominating-file
-   file
+   ;; We must not pass regular file here because
+   ;; `locate-dominating-file' will just pass it into its second
+   ;; lambda argument . But our lambda expects only directories.
+   (if (file-directory-p file)
+       file
+     (file-name-directory file))
```

Comment:
@fmdkdd ok; thanks for the explanation. @sergv I'm fine with that too, as long as we document the behavior (@fmdkdd, WDYT?)